### PR TITLE
GetBlockByNumber should use tx result data from GetBlockResults

### PR DESCRIPTION
**Pull request by atchapcyp**
ref : https://github.com/loomnetwork/loomchain/issues/1443

- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

### CODE REVIEW

File: eth/query/block.go [hunk 1]
```diff
@@ -81,12 +81,19 @@ func GetBlockByNumber(
 	blockInfo.LogsBloom = eth.EncBytes(bloomFilter)
 	for index, tx := range blockResult.Block.Data.Txs {
 		if full {
-			txResult, err := blockStore.GetTxResult(tx.Hash())
-			if err != nil {
-				return resp, errors.Wrapf(err, "cant find tx details, hash %X", tx.Hash())
+			var txResultBytes []byte
+			txResult, err := blockStore.GetBlockResults(&height)
```
Comment (pathornteng) [inline eth/query/block.go]: `height` does not change in the loop so `GetBlockResult` will return the same data, move this one out of `range blockResult.Block.Data.Txs` and change `txResult` to `blockResults`

File: eth/query/block.go [hunk 2]
```diff
@@ -79,14 +79,24 @@ func GetBlockByNumber(
 	blockInfo.Number = eth.EncInt(height)
 	bloomFilter := evmAuxStore.GetBloomFilter(uint64(height))
 	blockInfo.LogsBloom = eth.EncBytes(bloomFilter)
+	blockResults, err := blockStore.GetBlockResults(&height)
```
Comment (pathornteng) [inline eth/query/block.go]: We only need to call `GetBlockResults` if `full` is true.

File: eth/query/block.go [hunk 3]
```diff
@@ -79,14 +79,24 @@ func GetBlockByNumber(
...
+		return resp, err
+	}
 	for index, tx := range blockResult.Block.Data.Txs {
 		if full {
-			txResult, err := blockStore.GetTxResult(tx.Hash())
-			if err != nil {
-				return resp, errors.Wrapf(err, "cant find tx details, hash %X", tx.Hash())
+			var blockResultBytes []byte
+			if blockResults == nil {
+				blockResults, err := blockStore.GetTxResult(tx.Hash())
```
Comment (pathornteng) [inline eth/query/block.go]: For this line, change `blockResults` to  `txResult`

File: eth/query/block.go [hunk 4]
```diff
@@ -79,14 +79,28 @@ func GetBlockByNumber(
 	blockInfo.Number = eth.EncInt(height)
 	bloomFilter := evmAuxStore.GetBloomFilter(uint64(height))
 	blockInfo.LogsBloom = eth.EncBytes(bloomFilter)
+	var blockResults *ctypes.ResultBlockResults
+	if full {
+		blockResults, err = blockStore.GetBlockResults(&height)
+		if err != nil {
+			return resp, errors.Wrapf(err, "results for block %v", height)
+		}
```
Comment (enlight) [inline eth/query/block.go]: An error here means the `blockResults` couldn't be loaded, if you return immediately then `blockResults == nil` always be `false` and the code will never fall back to using `blockStore.GetTxResult(tx.Hash())`.

**Pull request merged**